### ISSUE TITLE
Adapt required json structure

### ISSUE THREAD

**Issue by tyzess** [lunactic/DIVAServices#17]
I generalized my application, so that almost all fields are now taken from the DIVAService. To do so, I've changed the response-jsons in 932b2ce75562214b3ca17c3bdb0e319a1df0cb22. The last thing to do is to change the validation and algorithm-creation/update to a more appropriate form. I'd propose the following:

{
    general: { `including all 'additional'-fields as well as name and description, see "conf/algorithmGeneral.json"` },
    input: { `same as before` },
    output: { `currently only contains the 'outputs'-field, see "conf/algorithmOutput.json"` },
      method: { `contains fields like 'language', 'executable_path', as well as 'file', see "conf/algorithmMethod.json"` }
}

@lunactic What do you think? For all changes see the commit 932b2ce75562214b3ca17c3bdb0e319a1df0cb22

**Comment by tyzess** [comment]
JSON strucuture looks currently like this:

```
{
  "general": {
    "name": "Canny Edge Detection",
    "description": "Who knows, I always only read candy detection ;)",
    "author": "DIVA UniFr",
    "email": "dev@diva.unifr.ch",
    "website": "http:\/\/www.unifr.ch",
    "type": "OCR"
  },
  "input": [
    {
      "inputImage": {

      }
    },
    {
      "outputFolder": {

      }
    }
  ],
  "output": {
    "output_type": "file"
  },
  "method": {
    "file": "http:\/\/localhost:3000\/uploads\/algorithm\/zip_file\/1\/algorithm.zip",
    "environment": "java:8",
    "language": "java",
    "executable_path": "cannyedgedetection\/cannyedgedetection.jar"
  }
}
```

**Comment by tyzess** [comment]
Maybe input: could be a hash instead of an array in the future?

**Comment by lunactic** [comment]
I think i solved it with 5ff4142 and 8022c86.

@uzysset can you test?

**Comment by tyzess** [comment]
Seems to work
